sign/ed25519/ed25519.go

```diff
@@ -1,144 +1,217 @@
 // Package ed25519 implements Ed25519 signature scheme as described in RFC-8032.
 //
-// References:
-//  - RFC8032 https://rfc-editor.org/rfc/rfc8032.txt
-//  - Ed25519 https://ed25519.cr.yp.to/
-//  - High-speed high-security signatures. https://doi.org/10.1007/s13389-012-0027-1
+// This package implements the three signature variants.
+//
+//  | Scheme Name | Sign Function     | Verification  | Context           |
+//  |-------------|-------------------|---------------|-------------------|
+//  | Ed25519     | Sign              | Verify        | None              |
+//  | Ed25519Ph   | SignPh            | VerifyPh      | Yes, can be empty |
+//  | Ed25519Ctx  | SignWithCtx       | VerifyWithCtx | Yes, non-empty    |
+//  | All above   | (PrivateKey).Sign | VerifyAny     | As above          |
+//
+// Specific functions for sign and verify are defined. A generic signing
+// function for all schemes is available through the crypto.Signer interface,
+// which is implemented by the PrivateKey type. A correspond all-in-one
+// verification method is provided by the VerifyAny function.
+//
+// Signing with Ed25519Ph or Ed25519Ctx requires a context string for domain
+// separation. This parameter is passed using a SignerOptions struct defined
+// in this package.
+//
+// Compatibility with crypto.ed25519
+//
+// These functions are compatible with the “Ed25519” function defined in
+// RFC-8032. However, unlike RFC 8032's formulation, this package's private
+// key representation includes a public key suffix to make multiple signing
+// operations with the same key more efficient. This package refers to the
+// RFC-8032 private key as the “seed”.
```

Comment:
Perhaps mention that this package can be used as a drop-in replacement of `crypto/ed25519`.